src/Sparrow/Platform/Posix/SmapsReader.cs

```diff
@@ -17,6 +72,7 @@ public class SmapsReader
 
         private const int BufferSize = 4096;
         private readonly byte[][] _smapsBuffer = {new byte[BufferSize], new byte[BufferSize]};
```

Comment:
The problem with allocating the `SmapsReader` is that it allocates 8KB at least, so we want to reduce this as much as possible.